src/core/pagination/relay.style.pagination.fn.ts

```diff
@@ -43,50 +41,89 @@ const DEFAULT_FIRST_VALUE = 5;
   }
 
  * !!! The pagination algorithm provided in the link about is not strictly followed !!!
+ * @param query A *SelectQueryBuilder*. You can provided it from  from the entity's repository class
+ * @param paginationArgs
+ * @param cursorColumn
  */
-export const getRelayStylePaginationResults = <T extends IBaseAlkemio>(
-  list: T[],
-  first?: number,
-  after?: string
-): IRelayStylePaginatedType<T> => {
-  let useFirst = DEFAULT_FIRST_VALUE;
-
-  if (first) {
-    if (first < 0) {
-      throw new PaginationInputOutOfBoundException(
-        'Non-negative integer expected for parameter "first"',
-        LogContext.COMMUNITY
-      );
-    }
-
-    useFirst = first;
+// todo: simplify; break in smaller functions; fix jsdoc
+export const getRelayStylePaginationResults = async <
+  T extends IBaseAlkemio & Paginationable
+>(
+  query: SelectQueryBuilder<T>,
+  paginationArgs: PaginationArgs,
+  cursorColumn = DEFAULT_CURSOR_COLUMN
+): Promise<IRelayStylePaginatedType<T>> => {
+  const logger = new Logger('Pagination');
```

Comment:
Why a different approach to logging? If it is due to this being a function + not able to inject then we may want to put this into a service etc? And then there should be a LogContext called Pagination etc.